### PR TITLE
Remove `required` from yamls and associated functions

### LINKED ISSUE #1936: QC: Remove `required` field from specs and `checkSpec()`/`CombineSpec()`/`Ingest()`
## QC Details

Per discussion with a2ai team, we are to remove the required field from all specs to avoid confusion

### LINKED ISSUE #1948: QC: Revisit "experimental" tags on functions
## QC Details

Currently, we are still marking MetricDetails and FlagOverTime functions as `experimental`, but they are included in our standard reports. Consider updating these to badges `stable`

### PR DESCRIPTION
**Pull request by lauramaxwell**
## Overview

Remove the required field from yaml specs and assume that all columns specified in the spec are required. Adjusts `checkSpec()` and `CombineSpec()` appropriately.

Also changes experimental flags to stable for FlagOverTime and report table creation functions

## Connected Issues

- Closes #1936
- Closes #1948

### CODE REVIEW

File: R/util-CombineSpecs.R [hunk 1]
```diff
@@ -11,23 +11,23 @@
 #' @examples
 #' spec1 <- list(
 #'   df1 = list(
-#'     col1 = list(required = TRUE),
-#'     col2 = list(required = TRUE)
+#'     col1 = list(type = character),
```
Comment (samussiah) [inline R/util-CombineSpecs.R:14]: Add quotes here?

File: R/util-checkSpec.R [hunk 2]
```diff
@@ -81,7 +81,7 @@ CheckSpec <- function(lData, lSpec) {
     }
     # check that required exist in data, if _all required is not specified
     if (!isTRUE(lSpec[[strDataFrame]]$`_all`$required)) {
```
Comment (samussiah) [inline R/util-checkSpec.R:83]: Keeping `required` on `_all`?

File: R/util-checkSpec.R [hunk 3]
```diff
@@ -14,24 +14,24 @@
...
-#' lData <- list(reporting_groups = gsm::reportingGroups, reporting_results = gsm::reportingResults)
+#' lData <- list(reporting_groups = gsm::reportingBounds, reporting_results = gsm::reportingResults)
 #' lSpec <- list(
-#'   reporting_groups = list(
-#'     GroupID = list(required = TRUE),
-#'     GroupLevel = list(required = TRUE),
-#'     Param = list(required = TRUE),
-#'     Value = list(required = TRUE)
+#'   reporting_bounds = list(
+#'     Threshold = list(type = numeric),
```
Comment (samussiah) [inline R/util-checkSpec.R:20]: Add quotes here too.

**Review by samussiah — CHANGES_REQUESTED** [review]
This makese sense to me - noticed a few unquoted data types.  I'm wondering if the `_all$required` scenario needs to be reconsidered.  It basically just tells `ApplySpec` to keep everything, but the specification is a little esoteric.  That said, if it works it works.

File: inst/workflow/3_reporting/Groups.yaml [hunk 4]
```diff
@@ -3,16 +3,16 @@ meta:
   ID: Groups
   Description: Reporting Groups Data
   Priority: 1
-spec: 
+spec:
   Mapped_STUDY:
     _all:
-      required: true
```
Comment (zdz2101) [inline inst/workflow/3_reporting/Groups.yaml:9]: were these meant to be removed? And does it need to be done on:

3_reporting/Bounds.yaml
4_modules/report_kri_country.yaml
4_modules/report_kri_site.yaml

inst/examples/2_AdverseEventWorkflow.R

?

Comment (lauramaxwell) [reply inst/workflow/3_reporting/Groups.yaml:9]: we went back and forth, but for the moment, we are leaving the `_all` specs as is, and will handle them later via #1952

Comment (lauramaxwell) [reply inst/workflow/3_reporting/Groups.yaml:9]: oh nice- thanks for catching that!

**Review by zdz2101 — APPROVED** [review]
LGTM, depending on which PR we merge first, the additional removal of required: true for study yaml will be needed as well

**Comment by lauramaxwell** [comment]
> LGTM, depending on which PR we merge first, the additional removal of required: true for study yaml will be needed as well

Thanks, @zdz2101 - i'm going to merge this in first and then remove the `required: true` fields for #1957

**Review by samussiah — APPROVED** [review]
Very thorough 💯

**Pull request merged**